app/user.go

```diff
@@ -43,26 +45,31 @@ func (a *App) CreateUserWithHash(user *model.User, hash string, data string) (*m
 		return nil, err
 	}
 
-	props := model.MapFromJson(strings.NewReader(data))
+	result := <-a.Srv.Store.Token().GetByToken(hash)
```

Comment:
Probably should rename the parameter `hash` to `tokenStr` or somthing.

Actually the function needs to be renamed. `CreateUserWithToken`?

Reply:
I create a new CreateUserWithToken but maintain the last one calling the new one for "backward compatibility", I don't know if anybody is using it. If you prefer it, I can remove the old function.